lib/OrdinaryDiffEqSDIRK/src/unified_sdirk_tableaus.jl

```diff
@@ -0,0 +1,757 @@
+                  T(0.18764102434672383)   -T(0.595297473576955)    T(0.9717899277217721)   γT                      0;
+                  T(0.490563388419108)     T(0.073570090080892)     T(0.4358665215)         T(0.0)                  γT]
+    
+    b = @SVector [T(0.490563388419108), T(0.073570090080892), T(0.4358665215), T(0.0), T(0.0)]
+    c = @SVector [γc, 2γc, T2(1), T2(1), T2(1)]
+    
+    b_embed = A[5, :]
+    
+    SDIRKTableau(A, b, c, γT, 4;
+                 b_embed=b_embed, embedded_order=3,
+                 is_fsal=false, is_stiffly_accurate=true,
+                 is_A_stable=true, is_L_stable=true,
+                 predictor_type=:hermite)
+end
+
+function Hairer42Tableau_unified(::Type{T}=Float64, ::Type{T2}=Float64) where {T, T2}
+    γT = T(0.3995)
+    γc = T2(0.3995)
+    
+    A = @SMatrix [γT                       0                        0                       0                       0;
+                  T(0.25)                  γT                       0                       0                       0;
+                  -T(0.08)                 0                        γT                      0                       0;
+                  T(0.19)                  -T(0.58)                 T(0.97)                 γT                      0;
+                  T(0.48)                  T(0.075)                 T(0.42)                 T(0.0)                  γT]
+    
+    b = @SVector [T(0.48), T(0.075), T(0.42), T(0.0), T(0.025)]
+    c = @SVector [γc, T2(0.6495), T2(0.9995), T2(0.98), T2(1.0)]
+    
+    b_embed = A[5, :]
+    
+    SDIRKTableau(A, b, c, γT, 4;
+                 b_embed=b_embed, embedded_order=3,
+                 is_fsal=false, is_stiffly_accurate=false,
+                 is_A_stable=true, is_L_stable=false,
+                 predictor_type=:hermite)
+end
+
+function CFNLIRK3Tableau_unified(::Type{T}=Float64, ::Type{T2}=Float64) where {T, T2}
+    γT = T(0.4358665215)
+    γc = T2(0.4358665215)
+    
+    A = @SMatrix [γT                       0                        0                       0;
+                  T(0.490563388419108)     γT                       0                       0;
+                  T(0.073570090080892)     0                        γT                      0;
+                  T(0.308809969973036)     T(1.490563388254106)     -T(1.235239879727145)   γT]
+    
+    b = @SVector [T(0.490563388419108), T(0.073570090080892), T(0.4358665215), T(0.0)]
+    c = @SVector [γc, 2γc, T2(1), T2(1)]
+    
+    b_embed = A[4, :]
+    
+    SDIRKTableau(A, b, c, γT, 3;
+                 b_embed=b_embed, embedded_order=2,
+                 is_fsal=false, is_stiffly_accurate=false,
+                 is_A_stable=true, is_L_stable=false,
+                 predictor_type=:hermite)
+end
+
+function get_sdirk_tableau(alg::Symbol, ::Type{T}=Float64, ::Type{T2}=Float64) where {T, T2}
```

Comment:
This is a bit wonky. It should instead just be like `SDIRKTableau(KenCarp4(), ...)` i.e. dispatch on the algorithm rather than different functions and this would all go away. But I can just refactor that at some later point. I wouldn't worry about this for now but it is a very weird way to do dispatch.